### DISCUSSION TITLE
More on the erosion coefficient

### DISCUSSION THREAD

**Discussion by AllanLopez** [wschwanghart/topotoolbox#38] [Q&A]
Dear Wolfgang,
Thank you for your response. After all these years I have learned to detect my mistakes in the execution, I hope.
What is obvious for you is martian language to me (also joking).

I guess a bridge is needed to fill the gap between Ksn extracted from chipplot and the following example given by the author below. I mean I have not worked anything; I do not know what to do.

Later I shall mention and idea to visit you for a couple of weeks next year.

All the best,

Allan 

for example,

Then, you can use fitlm or fitnlm to fit ksn and erosion rate and examine which fit explain the overall relationship.

Once you decide your functional form, you can calculate erosion coefficient using the best-fit parameters (e.g., power-law exponents)

xda1 = ksn; yda1 = erosion;

beta0 = [1];

nonlm0 = fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^1,beta0); %linear without intercept

 
beta0 = [1e-2 1];

nonlm1 = fitnlm(xda1, yda1 , @(b,xda1)exp(xda1.*b(1) +b(2)),beta0) ;%exponential

 beta0 = [1e-2 1];

nonlm2 =fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^b(2),beta0) ; %power law

 

beta0 = [1 0];

nonlm3 = fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^1+b(2),beta0); %linear with intercept

 
b_n2 = nonlm2.Coefficients{:,1}

erocoef_n2 = erosion./(ksn.^b_n2(2)); %n = 0.49, coefficient for nonlinear powerlaw fit

erocoef = erosion./(ksn); %n = 0.49, coefficient for linear fit without intercept

**Comment by wschwanghart** [comment]
Hi Allan,
well, you'll need basin-wide erosion rates which Seulgi named as vector ```erosion```.
Then, you'll have to calculate basin-averaged ksn values. [Richard Ott](https://github.com/Richard-Ott) explains [here](https://topotoolbox.wordpress.com/2021/07/14/mean-basin-ksn-and-smoothing-madness/) how to do this. Also, these values should be stored as a vector and must have the same length as the vector ```erosion```. Then, there are the options detailed by Seulgi. In the simplest case, there is a linear relation between both variables so that you can do it like she does:
```
xda1 = ksn; yda1 = erosion;
beta0 = [1];
nonlm0 = fitnlm(xda1, yda1 , @(b,xda1)b(1).*xda1.^1,beta0); %linear without intercept
```
In fact, this can also be done by
```
beta = xda1\yda1;
```
This gives you a scalar value, which is the erosion coefficient. So, no need to write anything as a shapefile. The result is just a single value which you retrieved from multiple basin-wide averaged denudation rates and basin-wide averaged ksn values.

Cheers, Wolfgang

**Comment by AllanLopez** [comment]
Dear Wolfgang,

Thank you very much indeed for your kind assistance and guidance into the
erosion affair.

All the best,

Allan
